addons/events/fnc_addPlayerEventHandler.sqf

```diff
@@ -121,115 +135,121 @@ if (_id != -1) then {
         GVAR(oldGroup) = grpNull;
         GVAR(oldLeader) = objNull;
         GVAR(oldWeapon) = "";
+        GVAR(oldMuzzle) = "";
+        GVAR(oldWeaponMode) = "";
         GVAR(oldLoadout) = [];
         GVAR(oldLoadoutNoAmmo) = [];
         GVAR(oldVehicle) = objNull;
-        GVAR(inVehicle) = false;
         GVAR(oldTurret) = [];
         GVAR(oldVisionMode) = -1;
         GVAR(oldCameraView) = "";
         GVAR(oldFeatureCamera) = "";
         GVAR(oldVisibleMap) = false;
-        GVAR(oldUAVControl) = [];
-        GVAR(controlledEntity) = objNull;
+        GVAR(oldState) = [];
 
         GVAR(playerEHInfo) pushBack addMissionEventHandler ["EachFrame", {call FUNC(playerEH_EachFrame)}];
         [QFUNC(playerEH_EachFrame), {
             SCRIPT(playerEH_EachFrame);
-            private _player = missionNamespace getVariable ["bis_fnc_moduleRemoteControl_unit", player];
-            if !(_player isEqualTo GVAR(oldUnit)) then {
-                [QGVAR(unitEvent), [_player, GVAR(oldUnit)]] call CBA_fnc_localEvent;
-                GVAR(oldUnit) = _player;
-                GVAR(oldUAVControl) = []; // force update
-            };
-
-            private _data = group _player;
-            if !(_data isEqualTo GVAR(oldGroup)) then {
-                [QGVAR(groupEvent), [_player, GVAR(oldGroup)]] call CBA_fnc_localEvent;
-                GVAR(oldGroup) = _data;
-            };
+            private _unit = missionNamespace getVariable ["bis_fnc_moduleRemoteControl_unit", player];
+            private _controlledEntity = _unit;
 
-            _data = leader _player;
-            if !(_data isEqualTo GVAR(oldLeader)) then {
-                [QGVAR(leaderEvent), [_player, GVAR(oldLeader)]] call CBA_fnc_localEvent;
-                GVAR(oldLeader) = _data;
+            private _uavControl = UAVControl getConnectedUAV _unit;
+            _uavControl = _uavControl param [(_uavControl find _unit) + 1, ""];
+            if (_uavControl isEqualTo "DRIVER") then {
+                _controlledEntity = driver getConnectedUAV _unit;
             };
 
-            _data = currentWeapon _player;
-            if !(_data isEqualTo GVAR(oldWeapon)) then {
-                GVAR(oldWeapon) = _data;
-                [QGVAR(weaponEvent), [_player, _data]] call CBA_fnc_localEvent;
+            if (_uavControl isEqualTo "GUNNER") then {
+                _controlledEntity = gunner getConnectedUAV _unit;
             };
 
-            _data = getUnitLoadout _player;
-            if !(_data isEqualTo GVAR(oldLoadout)) then {
-                GVAR(oldLoadout) = _data;
+            private _state = [
+                _unit, group _unit, leader _unit,
+                currentWeapon _unit, currentMuzzle _unit, currentWeaponMode _unit,
+                getUnitLoadout _unit,
+                vehicle _unit, _unit call CBA_fnc_turretPath,
```

Comment:
this func is a little expensive, could also add an exitWith to turretPath
```suggestion
                vehicle _unit, if (vehicle _unit != _unit) then { _unit call CBA_fnc_turretPath } else { [] },
```

Reply:
I am thinking of copying the turret getter function into the code to save the `call` and do more specific optimizations to it. The currentUnit function is hard coded into this as well instead of called, presumably to save one call and one params.